acceptance/tests/environment/negative/agent_run_should_fail_if_env_unreadable.rb

```diff
@@ -0,0 +1,57 @@
+test_name "C97899 - Agent run should fail if environment is unreadable" do
+
+  testdir = ''
+  env_path = ''
+  test_env = ''
+
+  step 'setup environments' do
+    testdir = create_tmpdir_for_user master, 'c97899_unreadable_envdir'
+    env_path = "#{testdir}/environments"
+    test_env = "#{env_path}/testing"
+
+    apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
+      File {
+        ensure => directory,
+        mode => "0770",
+        owner => #{master.puppet['user']},
+        group => #{master.puppet['group']},
+      }
+      file {
+        '#{env_path}':;
+        '#{env_path}/production':;
+        '#{test_env}':;
+        '#{test_env}/manifests':;
+        '#{test_env}/modules':;
+      }
+      file { '#{test_env}/manifests/site.pp':
+        ensure => file,
+        mode => "0640",
+        content => 'node default { notify { "Hello agent": } }',
+      }
+    MANIFEST
+  end
+
+  step 'change permissions of envdir to 644' do
+    on(master, "chmod 644 #{test_env}")
```

Comment:
@james-stocks , I am unable to reproduce the case where the `site.pp` file is too restrictive. I have added another test case to this PR for that scenario. It currently, fails the `expect_failure` condition when I run it against a recent master puppet-agent (SHA=e15c8b567fee6846f06aa8f0e74a07dd8fe90441).